mne/io/brainvision/brainvision.py

```diff
@@ -513,9 +535,40 @@ def _get_vhdr_info(vhdr_fname, eog, misc, scale, montage):
                 if lp_s:
                     info['lowpass'] = 1. / info['lowpass']
         else:
-            info['lowpass'] = np.min(np.array(lowpass, dtype=np.float))
-            warn('Channels contain different lowpass filters. Lowest filter '
-                 'setting will be stored.')
+            heterogeneous_lp_filter = True
+            if lp_s:
+                # We convert channels with disabled filters to having
+                # infinitely relaxed / no filters
+                lowpass = [float(_) if _ not in ('NaN', 'Off') else 0.0
+                           for _ in lowpass]
+                info['lowpass'] = np.max(np.array(lowpass, dtype=np.float))
+                try:
+                    info['lowpass'] = 1. / info['lowpass']
+                except ZeroDivisionError:
+                    # No lowpass actually set (always Off or equivalent)
+                    # so we let lowpass be set in _empty_info
+                    del info['lowpass']
+                    # not actually heterogeneous in effect
+                    # ... just heterogeneously disabled
+                    heterogeneous_lp_filter = False
+            else:
+                # We convert channels with disabled filters to having
+                # infinitely relaxed / no filters
+                lowpass = [float(_) if _ not in ('NaN', 'Off') else np.Inf
+                           for _ in lowpass]
```

Comment:
Do you mean single-character variable names? And `_` is often used as a throwaway variable in list comprehensions too, which is why it's used here -- it's not interesting after the comprehension has been evaluated.